### PR TITLE
DeserializationButtonFeature

### PR DESCRIPTION
**Pull request by EugenKoulik**

### CODE REVIEW

File: Forms/StartForm/LoadFileButton.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        }
+
+        private DeserializeGraph deserializeGraph = new DeserializeGraph();
+
+        public Graph LoadGraph { get; set; }
+
+        public StartForm.StartForm StartForm;
+
+        public InputCountVertexForm NextForm;
```
Comment (dnbss) [inline Forms/StartForm/LoadFileButton.cs:33]: Variables start with a capital letter

File: Forms/StartForm/LoadFileButton.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+            Size = new System.Drawing.Size(100, 50);
+
+            Anchor = (AnchorStyles.Bottom | AnchorStyles.Right); // anchorage to place
+
+            Click += new EventHandler(ButtonClick);
+
+        }
+
+        private DeserializeGraph deserializeGraph = new DeserializeGraph();
```
Comment (dnbss) [inline Forms/StartForm/LoadFileButton.cs:27]: I think initialization should be done in the constructor

Comment (dnbss) [inline Forms/StartForm/LoadFileButton.cs]: I think initialization should be done in the constructor

File: Forms/StartForm/LoadFileButton.cs (hunk 1 above)
Comment (dnbss) [inline Forms/StartForm/LoadFileButton.cs]: Variables start with a capital letter

File: SerializingTest/GraphConversionTest.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+            vertex.Nodes.Add(new Node() { Weight = 228, Connectable = 20 });
+            vertex.Nodes.Add(new Node() { Weight = 14, Connectable = 17 });
+
+            graph.Vertexs.Add(new Vertex() { Nodes = vertex.Nodes, Id = 42 });
+
+        }
+
+        [Test]
+        public void SerializationEdgeWeight_Weight_228return()
```
Comment (dnbss) [inline SerializingTest/GraphConversionTest.cs]: After the name of the test method, the word "Test"should be written. Example: SomeMethodTest_Scenario_ExpectedResult

File: SerializingTest/GraphConversionTest.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+using CraphModel;
+using System.Collections.Generic;
+using Serializing;
+
+
+
+namespace SerializingTest
+{
+    public class GraphConversionTest
+    {
```
Comment (Maksim-Trolina) [inline SerializingTest/GraphConversionTest.cs]: One test class tests one class.Test class name problems, see what other test classes are called

File: SerializingTest/GraphConversionTest.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            int expected = 228;
+            int actual = deserializeGraph.LoadGraph().Vertexs[0].Nodes[0].Weight;          
+
+            Assert.AreEqual(expected, actual);
+        }
+
+
+        [Test]
+        public void SerializationGraphId_Id_42return()
+        {
```
Comment (Maksim-Trolina) [inline SerializingTest/GraphConversionTest.cs]: The method name must not contain numbers.

File: Forms/StartForm/LoadFileButton.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+        private StartForm.StartForm startForm;
+
+        private List<VertexDraw> vertexDraws;
+
+        private List<EdgeDraw> edgeDraws;
+
+        private StartForm.DrawForm drawForm;
+
+        private FileInfo fI = new FileInfo("Graphs.json");
```
Comment (Maksim-Trolina) [inline Forms/StartForm/LoadFileButton.cs]: Bad way to initialize

Comment (Maksim-Trolina) [inline Forms/StartForm/LoadFileButton.cs]: It is better to make the filename permanent, as it is used in several places in the project.

**Review by Maksim-Trolina — APPROVED** [review]

**Review by dnbss — APPROVED** [review]

**Pull request merged**